sqlglot/dialects/db2.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+import typing as t
+from sqlglot import exp, generator, parser, tokens
+
+from sqlglot.dialects.dialect import Dialect
+
+# from sqlglot.generator import Generator
+from sqlglot.tokens import Tokenizer, TokenType
+
+from sqlglot.helper import seq_get
+
+
+from sqlglot.dialects.dialect import (
+    build_date_delta_with_interval,
+    unit_to_var,
+    build_formatted_time,
+    NormalizationStrategy,
+)
+
+
+def _build_timetostr_or_tochar(args: t.List) -> exp.TimeToStr | exp.ToChar:
+    this = seq_get(args, 0)
+
+    if this and not this.type:
+        from sqlglot.optimizer.annotate_types import annotate_types
+
+        annotate_types(this)
+        if this.is_type(*exp.DataType.TEMPORAL_TYPES):
+            return build_formatted_time(exp.TimeToStr, "db2", default=True)(args)
+
+    return exp.ToChar.from_arg_list(args)
+
+
+def date_add_sql(
+    kind: str,
+) -> t.Callable[[generator.Generator, exp.Expression], str]:
+    def func(self: generator.Generator, expression: exp.Expression) -> str:
+        return self.func(
+            f"DATE_{kind}",
+            expression.this,
+            exp.Interval(this=expression.expression, unit=unit_to_var(expression)),
+        )
+
+    return func
+
+
+class Db2(Dialect):
+    DATE_FORMAT = "'yyyy-MM-dd'"
+    NORMALIZATION_STRATEGY = NormalizationStrategy.UPPERCASE
+
+    TIME_MAPPING = {
+        "AM": "%p",
+        "A.M.": "%p",
+        "PM": "%p",
+        "P.M.": "%p",
+        "D": "%u",
+        "DAY": "%A",
+        "DD": "%d",
+        "DDD": "%j",
+        "DY": "%a",
+        "HH": "%I",
+        "HH12": "%I",
+        "HH24": "%H",
+        "IW": "%V",
+        "MI": "%M",
+        "MM": "%m",
+        "MON": "%b",
+        "MONTH": "%B",
+        "SS": "%S",
+        "WW": "%W",
+        "YY": "%y",
+        "YYYY": "%Y",
+        "FF": "%f",
+        "FF1": "%f",
+        "FF2": "%f",
+        "FF3": "%f",
+        "FF4": "%f",
+        "FF5": "%f",
+        "FF6": "%f",
+    }
+
+    class Tokenizer(tokens.Tokenizer):
+        QUOTES = ["'", '"']
+        IDENTIFIERS = ["`"]
+
+        # Associates certain meaningful words with tokens that capture their intent
+        KEYWORDS = {
+            **Tokenizer.KEYWORDS,
+            "(+)": TokenType.JOIN_MARKER,
+            "INT64": TokenType.BIGINT,
+            "FLOAT64": TokenType.DOUBLE,
```

Comment:
This is repeated across multiple dialects, let's move it to the base tokenizer's `KEYWORDS` and get rid of the other entries.